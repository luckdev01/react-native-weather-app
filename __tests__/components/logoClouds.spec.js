import 'react-native';
import React from 'react';
import {Animated, Image, Text} from 'react-native';
import { shallow } from 'enzyme';
import LogoClouds from '../../src/components/logoClouds';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<LogoClouds />', () => {
  it('Input and Button component renders correctly', () => {
    const tree = renderer.create(
      <LogoClouds scale={1} rotate={'30deg'} />
    );
    const json = tree.toJSON();
    expect(json).toMatchSnapshot();
  });
  const wrapper = shallow(<LogoClouds scale={1} rotate={'30deg'} />);

  it('Should exists', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('Should have one Animated image', () => {
    expect(wrapper.find(Animated.Image)).toHaveLength(1);
  });

  it('Should have one image', () => {
    expect(wrapper.find(Image)).toHaveLength(1);
  });

  it('Should have one text', () => {
    expect(wrapper.find(Text)).toHaveLength(1);
  });
});
