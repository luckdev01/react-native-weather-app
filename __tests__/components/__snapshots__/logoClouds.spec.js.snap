exports[`<LogoClouds /> Input and Button component renders correctly 1`] = `
<View>
  <Image
    source={1}
    style={
      Object {
        "marginLeft": 0,
        "marginTop": 150,
        "tintColor": "#ffff00",
        "transform": Array [
          Object {
            "scale": 1,
          },
          Object {
            "rotate": "30deg",
          },
        ],
      }
    } />
  <Image
    source={1}
    style={
      Object {
        "marginLeft": 50,
        "marginTop": -142,
        "tintColor": "#F5FCFF",
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "Nunito-Bold",
        "fontSize": 50,
        "height": 70,
        "textAlign": "center",
      }
    }>
    KeradWeather
  </Text>
</View>
`;
